### ISSUE TITLE
Named timeout

### ISSUE THREAD

**Issue by asilvas** [caolan/async#1097]
https://github.com/caolan/async#timeoutfunction-miliseconds

When there are more than one location that can timeout, being able to track which is critical.

Please either add a `info` argument (string or object), or use the name of the callback function, in the error object. Ex: `err.info = info;`

**Comment by asilvas** [comment]
Exampel scenario:

```
async.auto({
  task1: async.timeout(function(cb) {
    cb();
  }, 1000),
  task2: ['task1', async.timeout(function(results, cb) {
    return; // naughty!
    cb();
  }, 1000)]
}, function(err, results) {
  if (err && err.code === 'TIMEDOUT') {
    // who failed?!
  }
}
```

**Comment by aearly** [comment]
I like the idea of using `function.name`, if available.  A configurable error message would also be useful.  How about a PR?

**Comment by asilvas** [comment]
Will do

**Comment by aearly** [comment]
Closed by #1098
